kwok/cloudprovider/helpers.go

```diff
@@ -39,12 +39,20 @@ var (
 	familyDelim = regexp.MustCompile(`[.-]`)
 )
 
+type KwokOfferings []KwokOffering
+
+type KwokOffering struct {
+	cloudprovider.Offering
+	// We wrap KwokOfferings with NodeSelectorRequirements for post-json processing translation
+	Requirements []v1.NodeSelectorRequirement
+}
+
 type InstanceTypeOptions struct {
-	Name             string                  `json:"name"`
-	Offerings        cloudprovider.Offerings `json:"offerings"`
-	Architecture     string                  `json:"architecture"`
-	OperatingSystems []v1.OSName             `json:"operatingSystems"`
-	Resources        v1.ResourceList         `json:"resources"`
+	Name             string          `json:"name"`
+	Offerings        KwokOfferings  `json:"offerings"`
```

Comment:
```suggestion
	Offerings        KWOKOfferings  `json:"offerings"`
```

nit: like other areas, go style is to retain the capitalization with acronyms